frontend/libs/studio-components/src/components/StudioTreeView/hooks/useTreeViewItemOpenOnHierarchySelect.ts

```diff
@@ -0,0 +1,18 @@
+import type { Dispatch, SetStateAction } from 'react';
+import { useEffect } from 'react';
+import { isDirectChildOfNode } from '../utils/domUtils';
+
+export const useTreeViewItemOpenOnHierarchySelect = (
+  rootId: string,
+  nodeId: string,
+  selectedId: string,
+  setOpen: Dispatch<SetStateAction<boolean>>,
+) => {
+  useEffect(() => {
+    if (nodeId === selectedId) {
+      setOpen(true);
+    } else if (isDirectChildOfNode(selectedId, rootId, nodeId)) {
+      setOpen(true);
+    }
+  }, [nodeId, rootId, selectedId, setOpen]);
```

Comment:
Not sure it is a better solution, but if we do like this (and pass children prop from `StudioTreeViewItem`) we do not need the extra isDirectChildOfNode method 🫣 

```suggestion
export const useTreeViewItemOpenOnHierarchySelect = (
  children: boolean,
  nodeId: string,
  selectedId: string,
  setOpen: Dispatch<SetStateAction<boolean>>,
) => {
  useEffect(() => {
    if (nodeId === selectedId && children) {
      setOpen(true);
    }
  }, [nodeId, selectedId, setOpen, children]);
};
```

Reply:
And I have not tested thoroughly, so take this suggestion with "a klype salt" 😅

Reply:
I think this would not solve #12038 unless i am misunderstanding something?

Reply:
Ah I see, I was not aware of that useCase. But then I agree!